crates/red_knot_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -222,8 +222,11 @@ impl SearchPaths {
         static_paths.push(stdlib_path);
 
         let site_packages_paths = match site_packages_paths {
-            SitePackages::Derived { venv_path } => VirtualEnvironment::new(venv_path, system)
-                .and_then(|venv| venv.site_packages_directories(system))?,
+            SitePackages::Derived { venv_path } => {
+                // TODO: Warn if venv python version isn't compatible.
```

Comment:
not sure I understand this TODO

Reply:
The idea here would be that we warn if `program.python_version` isn't compatible with the venv's python because that suggests a configuration error.

Reply:
I don't think that's necessarily true. People often use a newer Python version for local development than they use in production. That would result in the project's `requires-python` being set to e.g. `">= 3.9"` even though they're using Python 3.13 locally. We'd type-check assuming Python 3.9 (the version they're using in production), unless they explicitly override that on the command line

Reply:
With incompatible I mean if your python version is older than what's specified in requires python. Using a newer version isn't incompatible with requires python.

Or we warn if you set an explicit target version but your venv version doesn't match that version (it's not the same major/minor)

Reply:
I still worry that these could lead to noisy warnings that won't be useful to most users... but we can play it by ear and modulate it depending on the feedback we get from users, I suppose.

Could you make the TODO a bit more expansive so it's clear what we might want to warn about in the future?